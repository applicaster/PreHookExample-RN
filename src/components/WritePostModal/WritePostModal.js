import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'react-native';
import MediaDetailsScreen from '../MediaDetailsScreen';

class WritePostModal extends Component {
  render() {
    const { isWritePostModalVisible, toggleModal } = this.props;
    return (
      <Modal
        animationType={'fade'}
        transparent={false}
        visible={isWritePostModalVisible}
        onRequestClose={toggleModal}
      >
        <MediaDetailsScreen />
      </Modal>
    );
  }
}

WritePostModal.propTypes = {
  isWritePostModalVisible: PropTypes.bool,
  toggleModal: PropTypes.func,
};

export default WritePostModal;
