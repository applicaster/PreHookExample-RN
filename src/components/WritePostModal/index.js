import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { toggleModal } from '../../actions';
import {
  getWritePostModalVisibility,
} from '../../selectors';

import WritePostModal from './WritePostModal';

const mapStateToProps = state => ({
  isWritePostModalVisible: getWritePostModalVisibility(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleModal,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(WritePostModal);
