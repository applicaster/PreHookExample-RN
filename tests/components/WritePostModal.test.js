import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import WritePostModal from '../../src/components/WritePostModal/WritePostModal';

describe('WritePostModal', () => {
  test('component renders correctly when it is visible', () => {
    const html = renderer.create(
      <WritePostModal isWritePostModalVisible />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when it is invisible', () => {
    const html = renderer.create(
      <WritePostModal isWritePostModalVisible={false} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
