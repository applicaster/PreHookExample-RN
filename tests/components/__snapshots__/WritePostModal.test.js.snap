// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WritePostModal component renders correctly when it is invisible 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={undefined}
  transparent={false}
  visible={false}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Write Post Screen Goes Here
  </Text>
  <Image
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "uri": "feed_rn_close_button",
      }
    }
    style={
      Array [
        Object {
          "height": 25,
          "marginLeft": 5,
          "tintColor": "#FFFFFF",
          "width": 25,
        },
        Object {
          "left": 0,
          "position": "absolute",
          "top": 30,
          "zIndex": 5,
        },
        Object {
          "tintColor": "#FFFFFF99",
        },
      ]
    }
    testID={undefined}
  />
</Modal>
`;

exports[`WritePostModal component renders correctly when it is visible 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={undefined}
  transparent={false}
  visible={true}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    Write Post Screen Goes Here
  </Text>
  <Image
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    source={
      Object {
        "uri": "feed_rn_close_button",
      }
    }
    style={
      Array [
        Object {
          "height": 25,
          "marginLeft": 5,
          "tintColor": "#FFFFFF",
          "width": 25,
        },
        Object {
          "left": 0,
          "position": "absolute",
          "top": 30,
          "zIndex": 5,
        },
        Object {
          "tintColor": "#FFFFFF99",
        },
      ]
    }
    testID={undefined}
  />
</Modal>
`;
